### PR TITLE
Sort EnvironmentTreeItem children

### PR DESCRIPTION
**Pull request by wwlorey**
Fixes https://github.com/microsoft/vscode-azurestaticwebapps/issues/371

Screen Shot 2021-05-05 at 4 38 08 PM

### CODE REVIEW

File: src/tree/EnvironmentTreeItem.ts [hunk 1]
```diff
@@ -124,6 +120,10 @@ export class EnvironmentTreeItem extends AzureParentTreeItem implements IAzureRe
             children.push(this.appSettingsTreeItem, this.functionsTreeItem);
         }
 
+        if (this.inWorkspace) {
```
Comment (nturinski) [inline src/tree/EnvironmentTreeItem.ts:123]: image

😏 

I think we should just fix the `compareChildrenImpl` command.  If we do something like 

```
if (item1 instanceof GenericTreeItem && !(item2 instanceof GenericTreeItem)) {
            return -1;
        } else if (!(item1 instanceof GenericTreeItem) && item2 instanceof GenericTreeItem) {
            return 1;
        } else {
            return super.compareChildrenImpl(item1, item2);
        }
}
```

**Review by nturinski — COMMENTED** [review]
While this works, we should probably just sort

File: src/tree/EnvironmentTreeItem.ts (hunk 1 above)
Comment (ejizba) [reply src/tree/EnvironmentTreeItem.ts:123]: Our pattern for app service and functions is to sort this level of the tree manually based on what we think is most important for users, not alphabetically (which is what `super.compareChildrenImpl` would do). I think we should keep doing the same here.

I'm guessing you don't want "Actions" at the bottom, which is possible if you sort alphabetically and if/when we ever localize.

**Comment by wwlorey** [comment]
Here's the new order:
Screen Shot 2021-05-06 at 11 38 28 AM
Screen Shot 2021-05-06 at 11 39 10 AM

**Review by nturinski — APPROVED** [review]

**Pull request merged**